auraed/src/runtime/cell_service/executables/auraed/nested_auraed.rs

```diff
@@ -157,6 +161,10 @@ impl NestedAuraed {
                 let process = procfs::process::Process::new(pid)
                     .map_err(|e| io::Error::new(ErrorKind::Other, e))?;
 
+                // TODO: Remove this temporary solution to waiting for the nested auraed
+                //       to start listening on the socket
+                std::thread::sleep(Duration::from_secs(3));
```

Comment:
This is very scary to start bringing into a runtime daemon, even if its only temporary. 

Maybe we should remove this in this PR and bring in another PR where we use the discovery endpoint?

Reply:
Sure